tests/ecosystem/upgrade/test_upgrade_ocp.py

```diff
@@ -0,0 +1,164 @@
+import logging
+import json
+
+from ocs_ci.framework import Config
+from ocs_ci.ocs import ocp
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.framework.testlib import ManageTest
+from ocs_ci.ocs.cluster import CephCluster
+from ocs_ci.ocs.exceptions import CephHealthException
+
+logger = logging.getLogger(__name__)
+
+# TODO: add image type validation (ga to ga , nightly to nightly, newer than current etc.)
+
+
+class TestUpgradeOCP(ManageTest):
+    """
+    1. check cluster health
+    2. check OCP version
+    3. perform OCP upgrade
+    4. check all OCP ClusterOperators
+    5. check OCP version
+    5. monitor cluster health
+    """
+    ocp_o = ocp.OCP()
+
+    def test_upgrade_ocp(self):
+        """
+
+        Tests OCS stability when upgrading OCP
+
+        """
+        config = Config()
+        ceph_cluster = CephCluster()
+        ceph_cluster.enable_health_monitor()
+
+        target_image = config.UPGRADE['ocp_upgrade_version']
+        image_path = config.UPGRADE['ocp_upgrade_path']
+        self.cluster_operators = self.get_all_cluster_operators()
+        logger.info(f" oc version: {self.get_current_oc_version()}")
+        self.get_all_cluster_operators()
+        upgrade_operator_timeout = TimeoutSampler(
+            timeout=2700,
+            sleep=60,
+            func=self.check_upgrade_completed,
+            target_version=target_image
+        )
+
+        # Upgrade OCP
+
+        self.upgrade_ocp(image=target_image, image_path=image_path)
+
+        # Wait for upgrade
+        for ocp_operator in self.cluster_operators:
+            logger.info(f"Checking upgrade status of {ocp_operator}:")
+            ver = self.get_cluster_operator_version(ocp_operator)
+            logger.info(f"current {ocp_operator} version: {ver}")
+            upgrade_operator_timeout.wait_for_func_status(result=True)
+
+        # validate cluster status and health
+        ceph_cluster.disable_health_monitor()
+        if ceph_cluster.health_error_status:
+            CephHealthException(
+                f"During upgrade hit Ceph HEALTH_ERROR: "
+                f"{ceph_cluster.health_error_status}"
+            )
+
+    def get_all_cluster_operators(self):
+        """
+        Get all ClusterOperators names in OCP
+
+        Returns:
+            list: cluster-operator names
+
+        """
+        ocp_obj = ocp.OCP(kind='ClusterOperator')
+        operator_info = ocp_obj.get("-o name", out_yaml_format=False, all_namespaces=True)
+        operators_full_names = str(operator_info).split()
+        operator_names = list()
+        for name in operators_full_names:
+            splitted = name.split('/')
+            for part in splitted:
+                if part == 'clusteroperator.config.openshift.io':
+                    splitted.remove(part)
```

Comment:
You shouldn't remove values from the list while you are iterating over it. See for example here:
https://sopython.com/canon/95/removing-items-from-a-list-while-iterating-over-the-list/

Can you please give me some example of data you have here and you want to get from them? Maybe we can think of better solution here.

Reply:
will use String.lstrip() instead